### PR TITLE
V3 current task view

### PR DESCRIPTION
**Pull request by 13bit**

### CODE REVIEW

File: public-v2/src/app/tracker/tasks/basket/basket.component.ts [hunk 1]
```diff
@@ -47,6 +48,9 @@ export class BasketComponent implements OnInit {
 
   ngOnInit() {
     this.getBasket();
+
+    this.basketService.activeBoardItem$
+      .subscribe((boardItem) => this.currentBI = boardItem);
```
Comment (alexkorsun) [inline public-v2/src/app/tracker/tasks/basket/basket.component.ts]: use full naming - boardItem

File: public-v2/src/app/tracker/tasks/basket/basket.component.ts [hunk 2]
```diff
@@ -145,4 +149,8 @@ export class BasketComponent implements OnInit {
         this.toastr.error(JSON.parse(err._body).error.toString(), 'Something was wrong');
       });
   }
+
+  onChangeStatus(event) {
+    this.currentBI = event;
```
Comment (alexkorsun) [inline public-v2/src/app/tracker/tasks/basket/basket.component.ts]: use full naming - boardItem

File: public-v2/src/app/tracker/tasks/basket/basket.component.html [hunk 3]
```diff
@@ -25,16 +25,21 @@ <h4 style="margin: 10px 0 20px; line-height: 24px;"  *ngIf="basket">{{basket.tit
...
                  id="time"/>
-          <span class="input-group-btn">
-      </span>
+          <span class="input-group-btn"></span>
         </div>
 
       </form>
     </div>
+
+    <basket-current-task *ngIf="currentBI" [boardItem]="currentBI">
```
Comment (alexkorsun) [inline public-v2/src/app/tracker/tasks/basket/basket.component.html]: use full naming - boardItem

File: public-v2/src/app/tracker/tasks/basket-current-task/basket-current-task.component.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      this.pause(changes['boardItem'].previousValue);
+    }
+
+    this.task = changes['boardItem'].currentValue.item;
+
+    this.countTaskSpentTime(changes['boardItem'].currentValue);
+  }
+
+  ngOnInit() {
+    if (!this.boardItem) {
```
Comment (alexkorsun) [inline public-v2/src/app/tracker/tasks/basket-current-task/basket-current-task.component.ts]: if (this.boardItem) {
this.task = this.boardItem.item;
this.countTaskSpentTime(this.boardItem);
}

File: public-v2/src/app/tracker/tasks/basket-current-task/basket-current-task.component.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  spendTime;
+
+  constructor(private taskService: TaskService,
+              private boardItemService: BoardItemService,
+              private basketService: BasketService,
+              private boardItemSpentTimeService: BoardItemSpentTimeService) {
+  }
+
+  ngOnChanges(changes: { [boardItem: string]: SimpleChange }): void {
+    if (Object.keys(changes['boardItem'].previousValue).length > 1
```
Comment (alexkorsun) [inline public-v2/src/app/tracker/tasks/basket-current-task/basket-current-task.component.ts]: simplify - use _id

File: public-v2/src/app/tracker/services/board-item-spent-time.service.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      .update(boardItem)
+      .switchMap(result => this.taskService.updateTask(boardItem.item))
+  }
+
+  getBoardItemSpendTime(boardItem: TaskBoardItem): Observable <Moment> {
+    let spentTime = 0;
+    let lastInProgress = null;
+
+    boardItem.timeLog.forEach(log => {
+      if (log.status === 'in progress' && !lastInProgress) {
```
Comment (alexkorsun) [inline public-v2/src/app/tracker/services/board-item-spent-time.service.ts]: leave one if with `if (log.status === 'in progress') {`

**Review by alexkorsun — CHANGES_REQUESTED** [review]

File: public-v2/src/app/tracker/tasks/basket/basket.component.ts (hunk 1 above)
Comment (13bit) [reply public-v2/src/app/tracker/tasks/basket/basket.component.ts]: +

File: public-v2/src/app/tracker/tasks/basket/basket.component.ts (hunk 2 above)
Comment (13bit) [reply public-v2/src/app/tracker/tasks/basket/basket.component.ts]: +

File: public-v2/src/app/tracker/tasks/basket/basket.component.html (hunk 3 above)
Comment (13bit) [reply public-v2/src/app/tracker/tasks/basket/basket.component.html]: +

File: public-v2/src/app/tracker/tasks/basket-current-task/basket-current-task.component.ts (hunk 5 above)
Comment (13bit) [reply public-v2/src/app/tracker/tasks/basket-current-task/basket-current-task.component.ts]: +

**Review by alexkorsun — APPROVED** [review]

**Pull request merged**